src/components/menu/menu.svelte

```diff
@@ -0,0 +1,234 @@
+<script context="module" lang="ts">
+  import type { HTMLAttributes } from 'svelte/elements'
+  declare global {
+    namespace JSX {
+      interface IntrinsicElements {
+        'leo-menu-item': HTMLAttributes<HTMLElement> & {
+          // Note: This should line up with Reacts key type, but we don't want
+          // to depend on React in this layer, so we just define it manually.
+          key?: string | number | null
+          children?: any
+        }
+      }
+    }
+  }
+</script>
+
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte'
+  import clickOutside from '../../svelteDirectives/clickOutside'
+  import Floating from '../floating/floating.svelte'
+
+  export let isOpen = false
+  export let target: HTMLElement | undefined = undefined
```

Comment:
Optional: I'm not 100% sure if you need the `= undefined` - you might be able to write:

```suggestion
  export let target?: HTMLElement
```

Reply:
i don't see this in the current codebase and I get a syntax error